### PR TITLE
Refactor/auto gas

### PR DESCRIPTION
**Pull request by BurntVal**

### CODE REVIEW

File: packages/abstraxion-core/src/GranteeSignerClient.ts [hunk 1]
```diff
@@ -93,7 +100,7 @@ export class GranteeSignerClient extends SigningCosmWasmClient {
   public async signAndBroadcast(
     signerAddress: string,
     messages: readonly EncodeObject[],
-    fee: StdFee | "auto" | number,
+    fee: StdFee | "auto" | number = "auto",
```
Comment (justinbarry) [inline packages/abstraxion-core/src/GranteeSignerClient.ts:103]: We should keep this api inline with the cosmjs interface
```suggestion
    fee: StdFee | "auto" | number,
```

**Review by justinbarry — CHANGES_REQUESTED** [review]
One change, also need a change set

File: packages/abstraxion-core/src/GranteeSignerClient.ts [hunk 2]
```diff
@@ -111,7 +118,45 @@ export class GranteeSignerClient extends SigningCosmWasmClient {
...
+      const gasEstimation = await this.simulate(signerAddress, messages, memo);
+      const multiplier = typeof fee == "number" ? fee : gasAdjustment;
+      const gasPrice = GasPrice.fromString(gasPriceString);
+      // usedFee = calculateFee(Math.round(gasEstimation * multiplier), gasPrice);
+      const calculatedFee = calculateFee(gasEstimation, gasPrice);
+      let gas = Math.ceil(
+        parseInt(calculatedFee.gas) * multiplier + gasAdjustmentMargin,
+      ).toString();
+
+      const chainId = await this.getChainId();
```
Comment (justinbarry) [inline packages/abstraxion-core/src/GranteeSignerClient.ts:137]: Couldn't we pass this in when it is instantiated in the provider?

**Review by justinbarry — APPROVED** [review]
LGTM

**Review by justinbarry — APPROVED** [review]
IGTM

**Review by BurntNerve — APPROVED** [review]
Looks good

**Pull request merged**